src/sync.cpp

```diff
@@ -468,6 +519,11 @@ namespace librealsense
     {
         return a->get_frame_number() < b->get_frame_number();
     }
+    bool frame_number_composite_matcher::is_smaller_than( matcher * missing, frame_holder & a )
+    {
+        auto const & next_expected = _next_expected.at( missing );
```

Comment:
Are we somehow protected from an exception here?
`at`

Reply:
If something is malfunctioning, I'd rather get the exception. The `missing` was already used before. It should be OK.